### ISSUE TITLE
Landmarks and role directory

### ISSUE THREAD

**Issue by JayPanoz** [w3c/epubcheck#1010]
Any idea why I now get this error in 4.2.0-beta: 

```
/Users/Pan/BlitzFramework/Blitz_template/Blitz_epub3.epub/OEBPS/Text/nav.xhtml(24,80): Erreur lors de l'analyse du fichier: la valeur de l'attribut "role" n'est pas valide ; doit être égal à "doc-index", "doc-pagelist", "doc-toc" ou "navigation"
```

While the markup pretty much follows [Role Mapping Table](https://idpf.github.io/epub-guides/epub-aria-authoring/) to the letter?

**Comment by mattgarrish** [comment]
The `role` attribute is more restricted in what values can be used with it depending on the element it's used on. The rules are defined in the [ARIA in HTML document](https://www.w3.org/TR/html-aria/).

You may have the correct mapping for whatever term you've put on the `nav`, but it may not be a navigation role, in other words. What term is causing the error?

**Comment by rdeltour** [comment]
Yeah the role mapping table is to be taken with a grain of salt; it's more an indication of which `role` _may_ be used to actually add AT-supported semantics for a given `epub:type`, but it doesn't necessarily mean that the role is conforming on the element which carried for `epub:type`.

For conformance rules on where DPUB ARIA roles  (and other ARIA roles) are allowed in  HTML, you need to refer to [ARIA in HTML](https://www.w3.org/TR/html-aria/).

In your case, the `directory` role is a subclass of the `list` role, so yeah it's not allowed on elements that already have a different semantics (native semantics should not be overridden).

**Comment by rdeltour** [comment]
(damn @mattgarrish is fast!)

**Comment by rdeltour** [comment]
Closing this issue as `invalid`, but feel free to keep on discussing here!

**Comment by mattgarrish** [comment]
Ya, but apparently I can't read issue titles... d'oh!

I'm thinking maybe we should add a column to the mapping table to list where the roles are allowed.

**Comment by JayPanoz** [comment]
Guilty as charged. 

I moved it from `nav` to the nested list and it’s now OK. 

However, I’ve seen other mappings in the wild putting it on `nav` – and `cover-image` on `figure` instead of `img` for instance, especially authoring tool plugins, automated tools, etc. so yeah, it would be useful to have such a column to the mapping table. 

Now I’ll have to deal with `doc-glossary` because `dl` is a group and not a list, and is in a huge turmoil it seems.

**Comment by mattgarrish** [comment]
> Now I’ll have to deal with `doc-glossary` because `dl` is a group and not a list, and is in a huge turmoil it seems.

Ya, `doc-glossary` is technically a landmark role, so there are problems both in that ARIA document and in our current release (it's not supposed to be on `dl` but on a `section` containing a glossary, and the schemas right now don't allow it anywhere). The example in the DPUB-ARIA module was incorrect, and seems to have caused various confusion.

I've filed an issue to fix the ARIA doc here: https://github.com/w3c/html-aria/issues/128

And to fix the schemas here: https://github.com/validator/validator/pull/771

We've fixed epubcheck for the next release, though, so it will validate on `section`.
